packages/addon/lua/gelly/mods/blood-mod/init.lua

```diff
@@ -96,21 +128,40 @@ end
 hook.Add(
 	"GellyXDamage",
 	"gelly.builtin.blood-mod",
-	function(_, attacker, position, force, damage, type)
+	function(victim, attacker, position, force, damage, type)
 		if
-			not attacker:IsPlayer()
-			and not attacker:IsNPC()
-			and attacker:GetClass() ~= "prop_ragdoll"
+			not victim:IsValid() or
+			not victim:IsPlayer() and
+			not victim:IsNPC() and
+			not victim:IsRagdoll()
 		then
 			return
 		end
-		-- 8194 is the combination used in M9K for bullets
-		if type == 8194 then
-			type = 2
+
+		local bloodcolor = 0
+
+		if victim:GetBloodColor() ~= nil then
+			bloodcolor = victim:GetBloodColor()
 		end
+		
+		local oldabsorption = gellyx.presets.getActivePreset().Material.Absorption
+		local material = gellyx.presets.getActivePreset().Material
+
+		if bloodcolor == 0 then -- the blood is red >>> ALSO WHEN IT'S RAGDOLL SINCE THEY DON'T HAVE BLOOD COLOR <<<
+			material.Absorption = Vector(0.05, 0.5, 0.5)
+		elseif bloodcolor == 2 or bloodcolor == 4 then  -- the blood is yellow
+			material.Absorption = Vector(0.25, 0.3, 0.5)
+		else 											-- the blood is green
+			material.Absorption = Vector(0.3, 0.25, 0.5)
+		end 
```

Comment:
A few of comments on this:
- nit: `bloodcolor` should be camelCase as `bloodColor` (and same for `oldabsorption` -> `oldAbsorption`) to fit the code style of the project
- You're checking the value of `bloodcolor` against number literals. The enum [`BLOOD_COLOR`](https://wiki.facepunch.com/gmod/Enums/BLOOD_COLOR) should be used instead for self-documentation
- Rather than using if statements, it would be better to define these absorption values (or even entire materials) as a lookup table with the `BLOOD_COLOR` enum values as the keys (similar to my comment on the weapon blood configs)